src/App.js

```diff
@@ -11,17 +11,33 @@ import LoginSuccess from "./pages/login-success/LoginSuccess";
 function App() {
   const location = useLocation();
 
+  const initializeAppForLoggedInUser = async (token) => {
+    const response = await fetch('https://api.spotify.com/v1/me',{
+      headers: {
+        Authorization: `Bearer ${token}`
+      }
+    });
+
+    if (response.ok) {
+      const json = await response.json();
+
+      rootStore.userStore.setUserResponseData(json);
+    } else {
+      // TODO: make sure to do proper error handling
+    }
+  }
```

Comment:
Step 4: We need to now create an asynchronous function inside of `App.js` that will fetch a users data.

We went over each line of code in the past, and I believe you added notes in your own branch. Definitely try to study and understand what this function is doing. Feel free to ask for any clarification!

Reply:
Also note the function is `async` and the fetch call has the keyword `await` on line 15. Definitely do some research if you don't understand this. We can go over this more as well if needed!